### PR TITLE
p1: check-clang-tidy-p1

### PR DESCRIPTION
**Pull request by timlee0119**

### CODE REVIEW

**Review by skyzh — APPROVED** [review]
LGTM. I would suggest:

* Create a new CMake variable `P1_FILES`.
* Pass this variable to check-clang-p1
* At the same time, add a new target `submit-p1`, which calls the zip command and creates a zip, using the same variable.

File: CMakeLists.txt [hunk 1]
```diff
@@ -239,4 +239,15 @@ add_custom_target(check-clang-tidy-p0
         "src/primer/p0_trie.cpp"
         )
 
+add_custom_target(check-clang-tidy-p1
+        ${BUSTUB_BUILD_SUPPORT_DIR}/run_clang_tidy.py                     # run LLVM's clang-tidy script
+        -clang-tidy-binary ${CLANG_TIDY_BIN}                              # using our clang-tidy binary
+        -p ${CMAKE_BINARY_DIR}                                            # using cmake's generated compile commands
+        -j 1
```
Comment (skyzh) [inline CMakeLists.txt:246]: Can we remove `-j1` here? I can't remember why I added it last time, but we absolutely don't want clang-tidy to run on only one core.

**Comment by timlee0119** [comment]
Added submit-p1 and fixed check-clang-tidy-p1

File: CMakeLists.txt [hunk 2]
```diff
@@ -235,8 +235,31 @@ add_custom_target(check-clang-tidy-p0
...
+set(P1_H_FILES
+        "src/include/container/hash/extendible_hash_table.h"
+        "src/include/buffer/lru_k_replacer.h"
+        "src/include/buffer/buffer_pool_manager_instance.h"
+        )
+add_custom_target(check-clang-tidy-p1
+        ${BUSTUB_BUILD_SUPPORT_DIR}/run_clang_tidy.py                     # run LLVM's clang-tidy script
+        -clang-tidy-binary ${CLANG_TIDY_BIN}                              # using our clang-tidy binary
+        -p ${CMAKE_BINARY_DIR}                                            # using cmake's generated compile commands
+        ${P1_CPP_FILES}
```
Comment (skyzh) [inline CMakeLists.txt:255]: clang-tidy will automatically ignore .h files, so we can use the same variable (which contains all source + header).

File: CMakeLists.txt [hunk 3]
```diff
@@ -235,8 +235,31 @@ add_custom_target(check-clang-tidy-p0
...
+        )
+add_custom_target(submit-p1
+        zip project1-submission.zip
+        ${P1_CPP_FILES}
+        ${P1_H_FILES}
+        WORKING_DIRECTORY ../
```
Comment (skyzh) [inline CMakeLists.txt:261]: students may not run `cmake` in build directory. They can run this in `build/build/build`. Consider use `CMAKE_CURRENT_SOURCE_DIR`.

Comment (timlee0119) [reply CMakeLists.txt]: done:)

File: CMakeLists.txt (hunk 2 above)
Comment (timlee0119) [reply CMakeLists.txt]: aye sir

**Review by skyzh — APPROVED** [review]

**Comment by skyzh** [comment]
You may merge after CI pass, good work!

**Pull request merged**